src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -1314,6 +1316,78 @@ private void AddPythonEnginesOptions()
                 options.Add(item.Name);
             }
             PythonEnginesList = options;
+            PopulatePythonEnginePackageMap();
+        }
+
+        private void PopulatePythonEnginePackageMap()
+        {
+            var packageBinPaths = new List<string>();
```

Comment:
why are you doing all this assembly searching and loading here ?
PythonEngines are already loaded In the PythonEngineManager constructor and in the PackageManager.PackageLOader.

The `PythonEnginesList` from this class will always be kept up to date when `PythonEngineManager.Instance.AvailableEngines` is changed. We should be able to store  all the info we need in the PythonENgine classes when they are initialized.

Reply:
>why are you doing all this assembly searching and loading here ?

Had to do all this to avoid breaking changes, also the idea was to not depend on the extension/engine to tell what package they come from (at-least for me), if allowed to change PythonEngine, or add a dependency on the extension itself to declare which package they belong to, it would have been way simpler. I agree.